nomos-da/kzgrs-backend/src/common/mod.rs

```diff
@@ -96,21 +104,24 @@ impl AsRef<[Chunk]> for Column {
 }
 
 impl ChunksMatrix {
+    #[must_use]
     pub fn len(&self) -> usize {
         self.0.len()
     }
+    #[must_use]
     pub fn is_empty(&self) -> bool {
         self.len() == 0
     }
     pub fn rows(&self) -> impl Iterator<Item = &Row> + '_ {
         self.0.iter()
     }
     #[cfg(feature = "parallel")]
+    #[must_use]
     pub fn par_rows(&self) -> impl ParallelIterator<Item = &Row> + '_ {
         self.0.par_iter()
     }
     pub fn columns(&self) -> impl Iterator<Item = Column> + '_ {
```

Comment:
Why not `must_use` in this method?

Reply:
Co-pilot answer:

> The columns method in the ChunksMatrix implementation returns an iterator over Column objects. Clippy might not have suggested `#[must_use]` for this method because it is common for iterators to be used immediately in a method chain or passed directly to other functions. The #[must_use] attribute is more typically recommended for methods where the return value is crucial to be used, like computations or transformations.

So it seems to me Clippy runs some heurstics on the source code to understand what are the places where the result is easier to be "misused", and suggest to add a `#[must_use]` there.